prometheus-connector/src/main/java/org/apache/flink/connector/prometheus/sink/PrometheusSinkWriter.java

```
@@ -0,0 +1,267 @@
+                    LOG.warn(
+                            "{},{} {} (discarded {} time-series, containing {} samples)",
+                            response.getCode(),
+                            response.getReasonPhrase(),
+                            responseBody,
+                            timeSeriesCount,
+                            sampleCount);
+                    counters.inc(NUM_SAMPLES_NON_RETRIABLE_DROPPED, sampleCount);
+                } else if (RemoteWriteResponseClassifier.isRetriableErrorResponse(response)) {
+                    LOG.warn(
+                            "{},{} {} (after retry limit reached, discarded {} time-series, containing {} samples)",
+                            response.getCode(),
+                            response.getReasonPhrase(),
+                            responseBody,
+                            timeSeriesCount,
+                            sampleCount);
+                    counters.inc(NUM_SAMPLES_RETRY_LIMIT_DROPPED, sampleCount);
+                }
+            }
+
+            // Never re-queue requests
+            requestResult.accept(Collections.emptyList());
+        }
+
+        @Override
+        public void failed(Exception ex) {
+            LOG.warn(
+                    "Exception executing the remote-write (discarded {} time-series containing {} samples)",
+                    timeSeriesCount,
+                    sampleCount,
+                    ex);
+            counters.inc(NUM_SAMPLES_DROPPED, sampleCount);
+            counters.inc(NUM_WRITE_REQUESTS_PERMANENTLY_FAILED);
+        }
+
+        @Override
+        public void cancelled() {
+            LOG.warn(
+                    "Write request execution cancelled (discarded {} time-series containing {} samples)",
+                    timeSeriesCount,
+                    sampleCount);
+            counters.inc(NUM_SAMPLES_DROPPED, sampleCount);
+            counters.inc(NUM_WRITE_REQUESTS_PERMANENTLY_FAILED);
+        }
```

I wonder if we should also propagate the exception upwards here rather than swallowing it. For example, this might be triggered on interrupt / non-graceful shutdown of the async Http client

---

Done: always throws an exception on cancelled